src/riak_kv_vnode.erl

```diff
@@ -2383,12 +2383,39 @@ encode_and_put_no_sib_check(Obj, Mod, Bucket, Key, IndexSpecs, ModState,
                         false ->
                             ok
                     end,
-                    PutRet = Mod:put(Bucket, Key, IndexSpecs, EncodedVal,
-                                     ModState),
+
+                    %% Timeseries data arrives during handoff with
+                    %% sext-encoded local keys. Unless care is taken,
+                    %% those keys are doubly-encoded, so here we will
+                    %% identify them and unencode the key.
+                    %%
+                    %% Not the ideal solution but sufficient.
+                    PutRet = case timeseries_data(Key, Obj) of
+                                 true ->
+                                     RawKey = sext:decode(Key),
+                                     Mod:put(Bucket, RawKey, IndexSpecs, EncodedVal,
```

Comment:
Discussing w/ Torben this morning - we believe that using sync_put in this case is most likely a premature optimization and may end up causing us grief (scheduler collapse anyone?). Would much prefer to see this case statement removed and create a function that just gets the final key for put using the timeseries_data function and sext decoding:

```erlang
get_final_key(<<16,0,0,0, _Rest/binary>> = Key, Object) ->
    case riak_object:is_ts(Object)  of
        {true,_ } ->
           sext:decode(Key);
        _ ->
           Key
   end;
get_final_key(Key, _) ->
    Key.
```

@erikleitch have you made any final decisions about sync_put vs. put here?